### ISSUE TITLE
Deploy fails when Exemption exists but Resource / Resource Group has been removed

### ISSUE THREAD

**Issue by robsissons-contino** [Azure/enterprise-azure-policy-as-code#424]
We have an error with exemptions that still exist in Azure which have a scope of a Resource (i.e. WebApp) however the Resource Group containing this resource was deleted.

The removal of the Resource Group does not appear to gracefully remove the Exemption associated with the Resource that no longer exists and leaves a state where the exemption is still visible through Graph, but not any other method.

If we then remove the exemption from the CSV because we know the resource/scope no longer exists then we receive an error during deployment:

Error from EPAC deployment:

```
Remove-AzPolicyExemption: /usr/local/share/powershell/Modules/EnterprisePolicyAsCode/8.4.5/functions/Deploy-PolicyPlan.ps1:100
Line |
 100 |  …     $null = Remove-AzPolicyExemption -Id $id -Force -ErrorAction Stop …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | ResourceGroupNotFound : Resource group
     | 'my-RG-name' could not be found.
     | CorrelationId: 1887674f-6fe5-4f46-9a0c-78c820580515
```

Evidence of Exemption still existing when querying through AzGraph:

    `Search-AzGraph -Query "PolicyResources | where name == '{exemptionName}'"`

All other methods of querying this exemption fail for the same reason - "ResourceGroupNotFound". Tested the following methods:

`Get-AzPolicyExemption -Id "{exemptionId}`
`az policy exemption show --name {exemptionName} --scope {exemptionScope}`
`GET https://management.azure.com{{Exemption_Id}}?api-version=2022-07-01-preview`
Manual view in portal does not show Exemption

Due to the fact that EPAC gets all exemptions using AzGraph, this exemption wants to be removed by the Plan after being removed from our Exemptions CSV.

Because all methods of querying / updating the Exemption fail we are unable to manually/programmatically update or delete this exemption.

Has anyone else seen the same? What is the expected behaviour in this scenario?

Attached document below shows screenshots to support the investigation:
[azure-exemptions-fail-to-delete-epac.docx](https://github.com/Azure/enterprise-azure-policy-as-code/files/13450362/azure-exemptions-fail-to-delete-epac.docx)

**Comment by anwather** [comment]
I can replicate this using the steps above - it is not being updated in the resource graph. Will look at resolving this part internally.

ICM: 444603092

**Comment by robsissons-contino** [comment]
Many thanks @anwather 

This was also raised as a support request through the portal FYI

**Comment by anwather** [comment]
Thanks let us know of the outcome particularly if they mention raising an internal case and I can link the one I have raised.

**Comment by anwather** [comment]
@robsissons-contino - your case has been linked to the issue I raised

**Comment by robsissons-contino** [comment]
Great news, many thanks @anwather

**Comment by naveensushanth** [comment]
I'm facing same issue, and due to this our whole pipeline is failing, unable to use EPAC for policy deployment as well.
Please suggest if you have workaround at least for now until bug will fix
Thanks in advance,

**Comment by anwather** [comment]
Let me get in contact with the team who was fixing this and I'll update when I hear something

**Comment by anwather** [comment]
@naveensushanth it has been confirmed as fixed and I have tested with no issue - I would suggest raising a case with Microsoft as this has to do with the behaviour of the Azure Resource Graph and not necessarily with EPAC. 

For my testing I created an exemption on a resource group - verified it existed within the graph, removed the resource group and the exemption no longer appeared as a result in the graph query. There was a second exemption also scope to the same group which was removed at the same time.
